storage/libradb/src/librarian/mod.rs

```diff
@@ -60,22 +59,49 @@ impl Librarian {
         );
         let num_txns = to_version - from_version;
         let txn_iter = self
+            .db
             .transaction_store
             .get_transaction_iter(from_version, num_txns as usize)?;
         txn_iter.collect::<Result<Vec<_>>>()
     }
 
     pub fn get_txn_by_version(&self, version: Version) -> Result<Transaction> {
-        self.transaction_store.get_transaction(version)
+        self.db.transaction_store.get_transaction(version)
     }
 
     pub fn get_account_state_by_version(
         &self,
         address: AccountAddress,
         version: Version,
     ) -> Result<Option<AccountStateBlob>> {
-        self.state_store
+        self.db
+            .state_store
             .get_account_state_with_proof_by_version(address, version)
             .map(|blob_and_proof| blob_and_proof.0)
     }
+
+    pub fn scan_events_by_seq(
+        &self,
+        key: &EventKey,
+        from_seq: u64,
+        to_seq: u64,
+    ) -> Result<Vec<(Version, ContractEvent)>> {
+        ensure!(
+            to_seq >= from_seq,
+            "'from' sequence {} > 'to' sequence {}",
+            from_seq,
+            to_seq
+        );
+        Ok((from_seq..to_seq)
+            .step_by(MAX_LIMIT as usize)
```

Comment:
This is lame man, this is lame. 🥴 You can probably do better by implementing an iterator.